WalletWasabi.Packager/Program.cs

```diff
@@ -188,7 +188,7 @@ private static void CreateDigests()
 
 			string torOsxZip = Path.Combine(torDaemonsDir, "tor-osx64.zip");
 			IoHelpers.BetterExtractZipToDirectoryAsync(torOsxZip, tempDir).GetAwaiter().GetResult();
-			File.Move(Path.Combine(tempDir, "Tor", "tor"), Path.Combine(tempDir, "TorOsx"));
+			File.Move(Path.Combine(tempDir, "Tor", "tor.real"), Path.Combine(tempDir, "TorOsx"));
```

Comment:
@lontivero @nopara73 this bug prevented the upgrade on macOS because the digests were always the same for the file `tor`. On macOS `tor.real` is what matters. Does this make sense?

Reply:
### First: Confirm the we are creating the digest for tor instead of tor.real
```sh
lontivero@lontivero-MAX-G5 ~/G/W/W/TorDaemons  $ unzip tor-osx64.zip                                                                        Archive:  tor-osx64.zip
  inflating: Tor/libevent-2.1.6.dylib
  inflating: Tor/tor
  inflating: Tor/tor.real
lontivero@lontivero-MAX-G5 ~/G/W/W/TorDaemons  $ cd Tor 
lontivero@lontivero-MAX-G5 ~/G/W/W/T/Tor  $ cat ../digests.txt 
ab11056dbe80f56b91802ba48d7d5117221c70b37866ae7ea019b6c1a24b4518
a3dec353c1e9dc2e8281e1e0ef00d94afd79ab645c1c4c985becb696bc6730db
c4ee142394bf7a53e43ca86ab3c03e3712d85c15941588fc4e6e8f5c7a88c654
lontivero@lontivero-MAX-G5 ~/G/W/W/T/Tor  $ sha256sum tor
a3dec353c1e9dc2e8281e1e0ef00d94afd79ab645c1c4c985becb696bc6730db  tor
lontivero@lontivero-MAX-G5 ~/G/W/W/T/Tor  $ sha256sum tor.real 102fae72b863af38e5f56874a6ba01bf1094015140ffd368a6913338480271e8  tor.real
```

**Note:** tor is a script that doesn't need to change, the real binary is `tor.real` and that has to always change

### Second: Could not confirm the tor doesn't change from version to version

```sh
lontivero@lontivero-MAX-G5 ~/G/W/W/TorDaemons  $ git checkout v1.0                                                                                            lontivero@lontivero-MAX-G5 ~/G/W/W/TorDaemons  $ unzip tor-osx64.zip                                                                                          Archive:  tor-osx64.zip
  inflating: Tor/libevent-2.0.5.dylib
  inflating: Tor/tor
  inflating: Tor/tor.real
lontivero@lontivero-MAX-G5 ~/G/W/W/TorDaemons  $ cd Tor/                                                                                                       lontivero@lontivero-MAX-G5 ~/G/W/W/T/Tor  $ sha256sum tor
ca7c7f0018478edd002df9a7c624034729345852e6612c01c7304b23d212409f  tor
```

**Note:** tor changed between v1.0 to v.1.1.11.1

### Confirm that both tor and tor.real had changed together in the past

```sh
lontivero@lontivero-MAX-G5 ~/G/W/W/TorDaemons  $ git checkout v1.0                                                                                                             lontivero@lontivero-MAX-G5 ~/G/W/W/TorDaemons  $ unzip tor-osx64.zip                                                                                           
Archive:  tor-osx64.zip
  inflating: Tor/libevent-2.0.5.dylib
  inflating: Tor/tor
  inflating: Tor/tor.real
lontivero@lontivero-MAX-G5 ~/G/W/W/TorDaemons  $ sha256sum ./Tor/tor.real
d92676cc33aca254fcd73a29ef56369758d4f54f9815fb7720ed305bc65a0334  ./Tor/tor.real
lontivero@lontivero-MAX-G5 ~/G/W/W/TorDaemons  $ sha256sum ./Tor/tor
ca7c7f0018478edd002df9a7c624034729345852e6612c01c7304b23d212409f  ./Tor/tor
lontivero@lontivero-MAX-G5 ~/G/W/W/TorDaemons  $ rm -rf Tor                                                                                                     lontivero@lontivero-MAX-G5 ~/G/W/W/TorDaemons  $ git checkout v1.1.10                                                                                           lontivero@lontivero-MAX-G5 ~/G/W/W/TorDaemons  $ unzip tor-osx64.zip                                                                                                         Archive:  tor-osx64.zip
  inflating: Tor/libevent-2.1.6.dylib
  inflating: Tor/tor
  inflating: Tor/tor.real
lontivero@lontivero-MAX-G5 ~/G/W/W/TorDaemons  $ sha256sum ./Tor/tor.real
d9e1a9b9fcc68183872ee2c16c1ead3af42618d90e67fb50403b939aaf7b89e8  ./Tor/tor.real
lontivero@lontivero-MAX-G5 ~/G/W/W/TorDaemons  $ sha256sum ./Tor/tor
ca7c7f0018478edd002df9a7c624034729345852e6612c01c7304b23d212409f  ./Tor/tor
lontivero@lontivero-MAX-G5 ~/G/W/W/TorDaemons  $ rm -rf Tor                                                                                                   lontivero@lontivero-MAX-G5 ~/G/W/W/TorDaemons  $ git checkout v1.1.11.1                                                                                                      
lontivero@lontivero-MAX-G5 ~/G/W/W/TorDaemons  $ unzip tor-osx64.zip                                                                                                       Archive:  tor-osx64.zip
  inflating: Tor/libevent-2.1.6.dylib
  inflating: Tor/tor
  inflating: Tor/tor.real
lontivero@lontivero-MAX-G5 ~/G/W/W/TorDaemons  $ sha256sum ./Tor/tor
a3dec353c1e9dc2e8281e1e0ef00d94afd79ab645c1c4c985becb696bc6730db  ./Tor/tor
lontivero@lontivero-MAX-G5 ~/G/W/W/TorDaemons  $ sha256sum ./Tor/tor.real
102fae72b863af38e5f56874a6ba01bf1094015140ffd368a6913338480271e8  ./Tor/tor.real
``` 

--------------------

## Conclusion

* We were calculating the digest for an script (`tor`) instead of the binary (`tor.real`)
* It could be possible that `tor` does not change from version to version.
* Comparing different versions in wasabi I didn't find this problematic situation where `tor.real` changes while `tor` does not.

We were doing it wrong but it seems we have been lucky.

Reply:
In less words: This change is correct @molnard and I suspect we were lucky.

Reply:
tor on macOS is just a bash script file.
It was not updated by this
https://github.com/zkSNACKs/WalletWasabi/commit/1c592ae2b7adaa1a0855f24d9c4df5c5f00c2027#diff-4549bed4624ad06cf35ccd05f4d50f42

The hash remained the same, so installed clients still used the old tor. 

Later it was hacked by adding a random comment to the bash script. 

https://github.com/zkSNACKs/WalletWasabi/commit/b263adf806061ea8afd10bde85d448b3f1c908ec#diff-4549bed4624ad06cf35ccd05f4d50f42